src/ec/packages/ec-component-expandable/expandable.html.twig

```diff
@@ -0,0 +1,102 @@
+{#
+  Parameters:
+    - "id" (string) (default: '')
+    - "label_expanded" (string) (default: '')
+    - "label_collapsed" (string) (default: '')
+    - "label_collapsed" (string) (default: '')
+    - "expanded" (string) (default: '')
+    - "button" (predefined structure) :  Button component structure 
+    - "extra_classes" (string) (default: '')
+    - "extra_attributes" (array) (default: []): format: [
+        {
+          "name" (string) (default: ''),
+          "value" (string) (default: '')
+        },
+        ...
+      ]
+
+  Blocks:
+    - "content"      
+#}
+
+{# Internal properties #}
+
+{% set _id = id|default('') %}
+{% set _css_class = 'ecl-expandable' %}
+{% set _extra_attributes = "data-ecl-expandable='true'" %}
+
+{% set _label_expanded = label_expanded|default('') %}
+{% set _label_collapsed = label_collapsed|default('') %}
+{% set _aria_controls = _id ~ '-content' %}
+{% set _content = content|default('') %}
+
+{% set _expanded = expanded|default('') %}
+{% set _content_hidden = '' %}
+
+{# Internal logic - Process properties #}
+
+{% if extra_classes is defined and extra_classes is not empty %}
+  {% set _css_class = _css_class ~ ' ' ~ extra_classes %}
+{% endif %}
+
+{% if extra_attributes is defined and extra_attributes is not empty and extra_attributes is iterable %}
+  {% for attr in extra_attributes %}
+    {% set _extra_attributes = _extra_attributes ~ ' ' ~ attr.name ~ '="' ~ attr.value ~ '"' %}
+  {% endfor %}
+{% endif %}
+
+{# Print the result #}
+
+{% spaceless %}
+<div class="{{- _css_class -}}"{{- _extra_attributes|raw -}}>
+  {%
+    set _button_attributes = [
+      { name: 'aria-controls', value: _aria_controls },
+      { name: 'data-ecl-expandable-toggle', value: 'true' },
+      { name: 'data-ecl-label-expanded', value: _label_expanded },
+      { name: 'data-ecl-label-collapsed', value: _label_collapsed },
+    ]
+  %}
+
+  {% if _expanded is not empty %}
+    {% 
+      set _button_attributes = _button_attributes|merge([
+        { name: 'aria-expanded', value: _expanded }
+      ])
+    %}
+
+    {% if _expanded == 'true' %}
+      {% set _label = _label_expanded %}
+      {% set _content_hidden = '' %}
+    {% else %}
+      {% set _label = _label_collapsed %}
+      {% set _content_hidden = 'hidden' %}
+    {% endif %}
+
+  {% endif %}  
+
+  {%- include '../ec-component-button/button.html.twig' with {
+    label: _label,
+    variant: 'secondary',
+    type: 'button',
+    icon: {
```

Comment:
is it mandatory to declare the empty icon here?

Reply:
Good question! In fact, I don't even know why I've "enforced" an icon, it's an option item in the button. So I'll remove.